plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/refactoring/quick_fixes/AddTokenAndImportStatement.java

```diff
@@ -193,42 +189,54 @@ public void createTextEdit(ComputedInfo computedInfo) {
 
             if (groupInto != null && realImportHandleInfo != null) {
                 //let's try to group it
-                int endLine = groupInto.getEndLine();
-                IRegion lineInformation = document.getLineInformation(endLine);
-                String strToAdd = ", " + realImportHandleInfo.getImportedStr().get(0);
+                int endLineNum = groupInto.getEndLine(); // get the number of last line of import
+                int startLineOffset = document.getLineInformation(groupInto.getStartLine()).getOffset();
+                int endLineOffset = document.getLineInformation(endLineNum).getOffset();
+                String endLine = PySelection.getLine(document, endLineNum); // get the string of last line of import
+                String endLineWithoutComment = PySelection.getLineWithoutCommentsOrLiterals(endLine); // also get the string of last line, but without comments
+
+                // get the string of full import statement, from first line to the last
+                String fullImportStatement = document.get().substring(startLineOffset,
```

Comment:
Instead of doing `doc.get()` (which will create a full string with the whole document) to then get a substring, you should use `document.get(int offset, int length)`, that way it just needs to create a small string to start with (it can make a big difference on really big files).